conformance/utils/kubernetes/helpers.go

```diff
@@ -390,13 +424,16 @@ func conditionsMatch(t *testing.T, expected, actual []metav1.Condition) bool {
 	return true
 }
 
-func findConditionInList(t *testing.T, conditions []metav1.Condition, condName, condValue string) bool {
+func findConditionInList(t *testing.T, conditions []metav1.Condition, condName, condValue, condReason string) bool {
 	for _, cond := range conditions {
 		if cond.Type == condName {
-			// TODO(nathancoleman): Compare Reason
 			if cond.Status == metav1.ConditionStatus(condValue) {
-				return true
+				if cond.Reason == condReason {
```

Comment:
I'm seeing a couple of tests start failing with this PR that I think should still pass:

- httproute-disallowed-kind
- httproute-invalid-cross-namespace-parent-ref

These both use the [HTTPRouteMustHaveNoAcceptedParents function](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/kubernetes/helpers.go#L214), which asserts that a Route has either no Accepted condition, or an "Accepted: false" condition. 

At least in Contour's implementation, we're setting an "Accepted: false" condition in both of these scenarios. The problem arises because `HTTPRouteMustHaveNoAcceptedParents` [does not pass a reason](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/utils/kubernetes/helpers.go#L245-L248) when checking for an "Accepted: false" condition, so `findConditionsInList` expects the actual condition to have an empty reason.

Not sure what the best fix is here, but one option would be to consider an empty reason string to mean "match any reason" instead of "match an empty string"? Otherwise, we might have to plumb a reason through from the place where `HTTPRouteMustHaveNoAcceptedParents` is called.

Reply:
Yeah, I agree that having an empty string passed for Reason can mean "match any Reason", since Reason strings should not normally be empty.